### PR TITLE
New squeaky datum and plushes

### PR DESCRIPTION
**Pull request by ninjanomnom**
Adds a new component for making things squeak, used in:
clown shoes
mice
bike horns
plushes

Refactors carp plush to use the new /obj/item/toy/plush type

Adds new plushes

plushies

### CODE REVIEW

**Comment by ShizCalev** [comment]
Donk Corporation is a member of the syndicate. Nanotrasen probably wouldn't be too cool with this idea.

**Comment by praisenarsie** [comment]
@ShizCalev explain donk co and donkpockets then :^)

**Comment by ShizCalev** [comment]
@praisenarsie What do you mean explain donk co? I just did.

**Comment by praisenarsie** [comment]
yes but WHY ARE THERE DONKPOCKETS ON NT STATIONS.

**Comment by ShizCalev** [comment]
The hold the patent, probably a generic /s

Could make a retcon PR for that if you'd like :^)

**Comment by Xhuis** [comment]
New canon: donk pockets and Donk Co. shuttles are disliked by NT but tolerated due to their cheapness and them being better than any competition, including NT-brand shuttles.

**Comment by GunHog** [comment]
This PR disappoints me. I thought we would have a Donksoft shuttle where the walls, floor, and interior are made entirely of foam. I was also imagining it with a bunch of Donksoft toys.

**Comment by ninjanomnom** [comment]
get me sprites for donksoft toys and I will add some to every shuttle

**Comment by praisenarsie** [comment]
@ShizCalev ^
make it happen ;))))))))))))))))))


(jk you don't have to)

**Comment by ninjanomnom** [comment]
sprite request: narsie and ratvar plushes

**Comment by Keekenox** [comment]
https://i.imgur.com/wksHFQt.png

NOW WITH SPRINGY ARM ACTION AND LIGHT UP GEMS! PEW PEW SPROING!

**Comment by nicbn** [comment]
Make it Nanotrasen instead of Donk, Nanotrasen is the corporation that do research which allows the creation of the shuttle engines.

**Comment by Keekenox** [comment]
Pls make PR with ratvar and narsie plushies, I will make sprites.

**Comment by ninjanomnom** [comment]
I'm already going to do it here and add them to shuttles @Keekenox

**Comment by Keekenox** [comment]
REEE ATOMIZE YOUR PRS AAAAAAAAAAH

**Comment by Qustinnus** [comment]
Ninja gets lectured by keek on pr etiquette, what a fuckin day to be alive

**Comment by ninjanomnom** [comment]
It is atomized, this is a Donk Co fluff pr relating to shuttles. Adding merchandise into the shuttles as part of it makes sense.

**Comment by ninjanomnom** [comment]
okay *now* this may be overscoped a bit because I made a squeaky datum for making anything squeak.

It was important for plushies to put in shuttles okay

File: code/datums/components/squeek.dm [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/datum/component/squeak
+	var/list/squeak_sounds = list('sound/items/toysqueak1.ogg'=1, 'sound/items/toysqueak2.ogg'=1, 'sound/items/toysqueak3.ogg'=1)
```
Comment (Cyberboss) [inline code/datums/components/squeek.dm]: make this a static list and another, initially empty list

File: code/datums/components/squeek.dm [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	else
+		RegisterSignal(COMSIG_ATOM_ENTERED, .proc/play_squeak)
+	if(istype(parent, /obj/item))
+		RegisterSignal(COMSIG_ITEM_ATTACK, .proc/play_squeak)
+		RegisterSignal(COMSIG_ITEM_ATTACK_SELF, .proc/play_squeak)
+		RegisterSignal(COMSIG_ITEM_ATTACK_OBJ, .proc/play_squeak)
+
+/datum/component/squeak/proc/play_squeak()
+	var/atom/owner = parent
+	playsound(owner.loc, pickweight(squeak_sounds), 30, 1, -1)
```
Comment (Cyberboss) [inline code/datums/components/squeek.dm]: just playsound(parent, ...

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by ChangelingRain — DISMISSED** [review]
holy shit atomize your pr; make the shuttle fluff change a _different_ pr

**Comment by ninjanomnom** [comment]
yeah the shuttle change is the odd one out here though it's what all this came from lol, moment while I move it over

**Comment by ninjanomnom** [comment]
@Cyberboss WIP can be removed now

File: code/game/objects/items/plushes.dm [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	icon = 'icons/obj/plushes.dmi'
+	icon_state = "debug"
+	attack_verb = list("thumped", "whomped", "bumped")
+	w_class = WEIGHT_CLASS_SMALL
+	resistance_flags = FLAMMABLE
+	var/list/squeak_override //Weighted list; If you want your plush to have different squeak sounds use this
+
+/obj/item/toy/plush/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, squeak_override)
```
Comment (optimumtact) [inline code/game/objects/items/plushes.dm]: QDEL this on destroy?

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 4]
```diff
@@ -79,15 +80,12 @@
 	item_state = "clown_shoes"
 	slowdown = SHOES_SLOWDOWN+1
 	item_color = "clown"
-	var/footstep = 1	//used for squeeks whilst walking
 	pockets = /obj/item/storage/internal/pocket/shoes/clown
+	var/list/step_sounds = list('sound/effects/clownstep1.ogg'=1,'sound/effects/clownstep2.ogg'=1)
```
Comment (optimumtact) [inline code/modules/clothing/shoes/miscellaneous.dm]: the step sounds should live only in the datum, not on the shoes themself I think

File: code/game/objects/items/plushes.dm (hunk 3 above)
Comment (Cyberboss) [reply code/game/objects/items/plushes.dm]: No, datum/Destroy cleans up components

File: code/_onclick/item_attack.dm [hunk 5]
```diff
@@ -9,6 +9,7 @@
 
 // Called when the item is in the active hand, and clicked; alternately, there is an 'activate held object' verb or you can hit pagedown.
 /obj/item/proc/attack_self(mob/user)
+	SendSignal(COMSIG_ITEM_ATTACK_SELF, user)
 	return
```
Comment (Cyberboss) [inline code/_onclick/item_attack.dm]: Remove this trailing return

File: code/datums/components/squeek.dm [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	if(volume_override)
+		volume = volume_override
+	if(step_delay_override)
+		step_delay = step_delay_override
+
+	if(istype(parent, /atom))
+		RegisterSignal(COMSIG_ATOM_BLOB_ACT, .proc/play_squeak)
+		RegisterSignal(COMSIG_ATOM_HULK_ATTACK, .proc/play_squeak)
+		RegisterSignal(COMSIG_PARENT_ATTACKBY, .proc/play_squeak)
+	if(istype(parent, /atom/movable))
```
Comment (Cyberboss) [inline code/datums/components/squeek.dm]: Lines 29-34 can get indented again

File: code/datums/components/squeek.dm [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+		RegisterSignal(COMSIG_MOVABLE_CROSSED, .proc/play_squeak)
+		RegisterSignal(COMSIG_MOVABLE_COLLIDE, .proc/play_squeak)
+		RegisterSignal(COMSIG_MOVABLE_IMPACT, .proc/play_squeak)
+	else
+		RegisterSignal(COMSIG_ATOM_ENTERED, .proc/play_squeak)
+	if(istype(parent, /obj/item))
```
Comment (Cyberboss) [inline code/datums/components/squeek.dm]: Should be under the atom/movable check

File: code/datums/components/squeek.dm [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+		RegisterSignal(COMSIG_ITEM_ATTACK, .proc/play_squeak)
+		RegisterSignal(COMSIG_ITEM_ATTACK_SELF, .proc/use_squeak)
+		RegisterSignal(COMSIG_ITEM_ATTACK_OBJ, .proc/play_squeak)
+	if(istype(parent, /obj/item/clothing/shoes))
```
Comment (Cyberboss) [inline code/datums/components/squeek.dm]: should be under the obj/iitem check

File: code/game/objects/items.dm [hunk 9]
```diff
@@ -217,6 +217,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 	return message
 
 /obj/item/attack_self(mob/user)
+	. = ..() //Why the hell is this proc defined twice
```
Comment (Cyberboss) [inline code/game/objects/items.dm]: Make it a single definition please

File: code/game/objects/items/clown_items.dm [hunk 10]
```diff
@@ -107,37 +107,22 @@
...
+	var/list/honksounds = list('sound/items/bikehorn.ogg'=1)
+
+/obj/item/bikehorn/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, honksounds, 50)
 
 /obj/item/bikehorn/suicide_act(mob/user)
 	user.visible_message("<span class='suicide'>[user] solemnly points the horn at [user.p_their()] temple! It looks like [user.p_theyre()] trying to commit suicide!</span>")
-	playsound(src.loc, honksound, 50, 1)
+	playsound(src.loc, pickweight(honksounds), 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/clown_items.dm]: playsound(src

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/clothing/shoes/miscellaneous.dm (hunk 4 above)
Comment (ninjanomnom) [reply code/modules/clothing/shoes/miscellaneous.dm]: I'll switch this to the other option which is to only do this work in initialize, but the only option other than that is to have a different component subtype for anything that wants a different squeak sound.

File: code/game/objects/items/clown_items.dm [hunk 11]
```diff
@@ -125,6 +124,10 @@
 	honksounds = list('sound/items/airhorn2.ogg'=1)
 	origin_tech = "materials=4;engineering=4"
 
+/obj/item/bikehorn/airhorn/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/items/airhorn2.ogg'=1), 50)
```
Comment (Cyberboss) [inline code/game/objects/items/clown_items.dm]: So, you know this will delete the component created by /obj/item/bikehorn, right?

**Comment by Cyberboss** [comment]
Travis

File: code/modules/clothing/shoes/miscellaneous.dm (hunk 4 above)
Comment (optimumtact) [reply code/modules/clothing/shoes/miscellaneous.dm]: hey you're the expert on this

File: code/game/objects/items/clown_items.dm (hunk 11 above)
Comment (ninjanomnom) [reply code/game/objects/items/clown_items.dm]: yeah, it's meant to override the old one for a new sound

**Pull request merged**